junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertInstanceOf.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.api;
+
+import static org.junit.jupiter.api.AssertionUtils.*;
+
+import java.util.function.Supplier;
+
+import org.opentest4j.AssertionFailedError;
+
+/**
+ * {@code AssertInstanceOf} is a collection of utility methods that support
+ * asserting that a value is of a expected instance type -
+ * in other words, if it can be assigned to the expected type.
+ *
+ * @since 5.8
+ */
+class AssertInstanceOf {
+
+	private AssertInstanceOf() {
+		/* no-op */
+	}
+
+	static void assertInstanceOf(Class<?> expectedType, Object actualValue) {
```

Comment:
I think these and the other methods should return the casted value.
```suggestion
	static <T> T assertInstanceOf(Class<T> expectedType, Object actualValue) {
```

Reply:
Yeah, this feels more useful. Actual value is an `Object` and after we verified it is of type `T` we can use it as that.